force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +170,35 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List<Opportunity> oppList = new List<Opportunity>();
         // Loop through each Opportunity name
+        for (String oppName : oppNames) {
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+            Account[] acct = [select Id from Account where Name = :accountName];
+            
+            if(acct.size() == 0) {
+                acct = new Account[1];
+                acct[0] = new Account(Name = accountName);
+                insert acct[0];
+            }
+            Opportunity[] newOpp = [Select Id, Name, AccountId, StageName, CloseDate from Opportunity where Name = :oppName];
+            
+            if(newOpp.size() == 0) {
+                newOpp = new Opportunity[1];
+                newOpp[0] = new Opportunity(Name = oppName, AccountId = acct[0].Id, StageName = 'Prospecting', CloseDate = Date.today().addMonths(4));
+            }
+            else {
+                newOpp[0].Name = oppName;
+                newOpp[0].AccountId = acct[0].Id;
+                newOpp[0].StageName = 'Prospecting';
+                newOpp[0].CloseDate = Date.today().addMonths(4);
+            }
+
+            oppList.add(newOpp[0]);
+        }
+        
+        upsert oppList;
```

Comment:
You have a lot of SOQL and DML happening inside of a loop here. This can cause all kinds of issues.